src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx

```diff
@@ -0,0 +1,159 @@
+import { useCallback, useMemo } from 'react'
+import styled from 'styled-components/macro'
+import CowProtocolLogo from 'components/CowProtocolLogo'
+import { formatUnits, parseUnits } from '@ethersproject/units'
+import { CurrencyAmount } from '@uniswap/sdk-core'
+
+import { InvestTokenGroup, TokenLogo, InvestSummary, InvestInput } from '../styled'
+import { formatSmart } from 'utils/format'
+import Row from 'components/Row'
+import { CheckCircle } from 'react-feather'
+import { InvestOptionProps } from '.'
+import { ApprovalState } from 'hooks/useApproveCallback'
+import { useCurrencyBalance } from 'state/wallet/hooks'
+import { useActiveWeb3React } from 'hooks/web3'
+
+const RangeSteps = styled.div`
+  display: flex;
+  align-items: center;
+  width: 100%;
+  justify-content: space-between;
+`
+
+const RangeStep = styled.button`
+  background: none;
+  border: none;
+  font-size: 0.8rem;
+  cursor: pointer;
+  color: blue;
+  padding: 0;
+`
+
+const INVESTMENT_STEPS = [0, 25, 50, 75, 100]
+
+export default function InvestOption({ approveState, approveCallback, updateInvestAmount, claim }: InvestOptionProps) {
+  const { currencyAmount, price, cost: maxCost, investedAmount } = claim
+
+  const { account } = useActiveWeb3React()
+
+  const token = currencyAmount?.currency
+
+  const balance = useCurrencyBalance(account || undefined, token)
+
+  const handlePercentChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    console.log(event.target.value)
+  }
+
+  const handleStepChange = (value: number) => {
+    console.log(value)
+  }
+
+  const onMaxClick = useCallback(() => {
+    if (!maxCost || !balance) {
+      return
+    }
+
+    const amount = maxCost.greaterThan(balance) ? balance : maxCost
+    // store the value as a string to prevent unnecessary re-renders
```

Comment:
This might be a dumb reason and not really apply, as we depend on other stuff that are object instances.
I started with this but did not get around to clear it.

One reason to keep it though is to be closer to the input, not a lot of conversions on every keystroke